docs/source/guides/advanced_custom_primitives.ipynb

```diff
@@ -0,0 +1,233 @@
+{
+ "cells": [
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "Advanced Custom Primitives Guide\n",
+    "--------------------------------"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from featuretools.primitives import TransformPrimitive\n",
+    "from featuretools.tests.testing_utils import make_ecommerce_entityset\n",
+    "from featuretools.variable_types import NaturalLanguage, Numeric\n",
+    "import featuretools as ft\n",
+    "import numpy as np\n",
+    "import re"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "Primitives With Additional Arguments\n",
+    "====================================\n",
+    "\n",
+    "Some features require more advanced calculations than others. Advanced features usually entail additional arguments to help output the desired value.  With custom primitives, you can use primitive arguments to help you create advanced features.\n",
+    "\n",
+    "\n",
+    "String Count Example\n",
+    "~~~~~~~~~~~~~~~~~~~~\n",
+    "\n",
+    "Make a primitive to count the number of times a specific string value occurs inside a text. First, derive a new transform primitive class from :class:`TransformPrimitive`. The primitive will take in a text column as the input and return a numeric column as the output, so set the input type as :class:`NaturalLanguage` and the return type as :class:`Numeric`. The specific string value is the additional argument, so define it as a *keyword* argument inside :meth:`__init__`. Then, override :meth:`get_function` to return the implementation as a function."
```

Comment:
For this paragraph, I think it could use a little more explanation on why we're using this example / what we gain from the example before diving into the specific of how we create the class.